### ISSUE TITLE
Service account is not propagated to Medusa standalone deployment

### ISSUE THREAD

**Issue by c3-clement** [k8ssandra/k8ssandra-operator#1294]
**What happened?**

The Medusa standalone deployment is using the `default` service account.

For AWS role based auth, that means that the `default` service account needs to be properly annotated and needs to be properly bound with the AWS IAM role.
Therefore, any pods created without a SA specified will be granted with AWS permissions.
This against the least privilege principle.

**Did you expect to see something different?**

I expect the Medusa standalone deployment's service account to be set to the value `K8ssandraCluster.cassandra.serviceAccount` .

**How to reproduce it (as minimally and precisely as possible)**:

Create a k8ssandra cluster with medusa enabled and a non-default service account.

**Environment**

* K8ssandra Operator version:
1.15
* Kubernetes version information:
1.29
* Kubernetes cluster kind:
EKS

┆Issue is synchronized with this [Jira Story](https://datastax.jira.com/browse/K8OP-31) by [Unito](https://www.unito.io)
┆Issue Number: K8OP-31

**Comment by c3-clement** [comment]
> We'll most probably remove the Medusa standalone pod altogether shortly as part of [this issue](https://github.com/k8ssandra/k8ssandra-operator/issues/1066).

makes sense

**Comment by L1ghtman2k** [comment]
@adejanovski do you know what are the guidelines on removing medusa standalone pod after k8ssandra upgrade?

Would it be sufficient to cleanup just the deployment? (My expectation was for the operator to remove it, as part of an upgrade, but I can't seem to find docs suggesting that we should do the cleanup on our own)

**Comment by adejanovski** [comment]
Yes, I dropped the ball when implementing its removal. Just delete the deployment and it won't get recreated.
Sorry about this
